v3/src/models/data/formula-fn-registry.ts

```diff
@@ -43,25 +43,35 @@ const aggregateFnWithFilterFactory = (fn: (values: number[]) => number) => {
     let expressionValues = evaluateNode(expression, scope)
     if (filter) {
       const filterValues = evaluateNode(filter, scope)
-      expressionValues = expressionValues.filter((v: any, i: number) => !!filterValues[i])
+      expressionValues = expressionValues.filter((v: any, i: number) =>
+        // Non-truthy expression values should not be part of the aggregate function result following V2 behavior.
+        // E.g. empty cells should not be counted in mean() function.
+        isValueNonEmpty(v) && isValueTruthy(filterValues[i])
+      )
+    } else {
+      // Non-truthy expression values should not be part of the aggregate function result following V2 behavior.
+      // E.g. empty cells should not be counted in mean() function.
```

Comment:
Seems like this comment could be reduced to:
```suggestion
      // Empty cells should not be included in aggregate functions.
```

Reply:
Thanks, I've committed this one and replaced another copy of this comment.